charts/quickwit/values.yaml

```diff
@@ -95,6 +95,8 @@ searcher:
 
   serviceAnnotations: {}
 
+  serviceType: {}
```

Comment:
Instead of `serviceType` and `serviceAnnotations`, I'd try to remain consistant with the default values which are using this form:

```yaml
service:
  type: ClusterIP
  annotations: {}
```

So I'd do something like this:

```yaml
searcher:
  service:
     type: ClusterIP
     annotations: {}
```

Also I'd avoid to have an empty hash as a default value for the `service.type` when it's supposed to be a String value. The default value we want is `ClusterIP` and if we want to fallback by default to the `.Values.service.type` value, in this case we just have to comment the bloc in the default values file:

```yaml
searcher:
  service:
    # type: ClusterIP
    annotations: {}
```

Because `serviceAnnotations` were already present on some place, it requires to have a major bump of the version with an entry in the README.md indicating the breaking changes.

If we want to avoid the breaking change, I'll understand completely. In this case we should remove the default values of `serviceType` everywhere and just add a commented value like this instead:

```yaml
serviceAnnotations: {}
# serviceType: ClusterIP
```

Reply:
For the default value, I can illustrate with this test that there's a problem. Let's do this overrided value file:

```yaml
---
config:
  default_index_root_uri: s3://qw-indexes
  storage:
    s3:
      endpoint: https://fr-par.scw.cloud
      region: fr-par
      access_key_id: SCWXXXXXXXX
      secret_access_key: XXXXXXXX

environment:
  QW_METASTORE_URI: s3://qw-indexes

indexer:
  serviceType: NodePort
```

We got this warning:

```shell
$ helm template . --values values.yaml --values values2.yaml
coalesce.go:220: warning: cannot overwrite table with non table for quickwit.indexer.serviceType (map[])
```

So there is something to do to fix this anyway :)

Reply:
As per https://github.com/quickwit-oss/helm-charts/pull/83#discussion_r1598446245 I think we should keep using the `serviceAnnotations` and `serviceType` ? (my first draft was using the service.type :sweat_smile: ) 
Kay, makes sens for the default value, just need to comment every occurence of `serviceType: {}` to `# serviceType: ClusterIP` and we're good?

Reply:
> Kay, makes sens for the default value, just need to comment every occurence of serviceType: {} to # serviceType: ClusterIP and we're good?

Yeah good to me :+1:

Once it's done, I'll perform some tests on kind to check if everything works correctly :)

Reply:
pushed the fix to serviceType